city_scrapers/spiders/il_governors_state_university.py

```diff
@@ -0,0 +1,201 @@
+        time = " ".join(item[1].css("* ::text").getall()).lower()
+        clean_time = self._normalize_time(time)
+        if clean_time is not None:
+            return datetime.strptime(f"{clean_day} {clean_time}", "%B %d %Y %I:%M %p")
+        # fall back to midnight if no time specified
+        return datetime.strptime(clean_day, "%B %d %Y")
+
+    def _parse_end(self, item):
+        """Parse end datetime as a naive datetime object. Added by pipeline if None.
+        Not available for this website."""
+        return None
+
+    def _parse_time_notes(self, item):
+        """Parse any additional notes on the timing of the meeting"""
+        return ""
+
+    def _parse_all_day(self, item):
+        """Parse or generate all-day status. Defaults to False. Doesn't seem to occur
+        for this website, with the possible exception of the retreats, which aren't
+        quite all day"""
+        return False
+
+    def _parse_location(self, item):
+        """Parse or generate location."""
+        location = item[1].css("* ::text").getall()
+        # remove time if present
+        location[0] = re.sub(self.time_re, "", location[0])
+        location = [loc.strip() for loc in location if len(loc.strip()) > 0]
+        # no obvious way to differentiate location names from addresses other than
+        # presence of numbers. We'll assume that the first line is title-only if it
+        # contains no numbers, otherwise that it begins the address.
+        # If there is no name, just the address, we'll use the first line
+        # of the address as the name. If there is no address, use the Governors State
+        # University address as the address. If after removing the time there is no
+        # information at all, we'll use "Governors State University" as the name
+        name = "Governors State University"
+        address = "1 University Pkwy,\nUniversity Park, IL 60484"
+        if len(location) > 0:
+            name = location[0]
+        if re.search(r"\d", name):
+            address = "\n".join(location)
+        elif len(location) > 1:
+            address = "\n".join(location[1:])
+        # special case for covid -- make sure zoom meetings don't show the university
+        # address!
+        if ("zoom" in name.lower()) or ("zoom" in address.lower()):
+            address = "Zoom"
+            name = "Zoom"
+        elif "location" in name.lower():
+            address = name
+        return {
+            "address": address,
+            "name": name,
+        }
+
+    def _parse_links(self, item):
+        """Parse or generate links."""
+        links = []
+        # the links to the agenda, if present, are in the third and fourth columns
+        for col in [2, 3]:
+            for link_parent in item[col].xpath("a"):
+                link_ext = link_parent.css("::attr(href)").get()
+                if link_ext is not None:
+                    link = self.base_url + link_ext
```

Comment:
Small thing here, but instead of passing the `base_url` statically we can use the `response.urljoin` method to create the full URL path if `response` is passed to `_parse_links`